File: src/main/java/leetcode/medium/perfect_squares/SolutionTest.java
package leetcode.medium.perfect_squares;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class SolutionTest {

    Solution solution = new Solution();

    @Test
    void test1() {
        int n = 12;
        int expected = 3;
        assertThat(solution.numSquares(n)).isEqualTo(expected);
    }

    @Test
    void test2() {
        int n = 13;
        int expected = 2;
        assertThat(solution.numSquares(n)).isEqualTo(expected);
    }

}

File: src/main/java/leetcode/medium/h_index/SolutionTest.kt
package leetcode.medium.h_index

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {

    val solution = Solution()

    @Test
    fun test1() {
        val citations = intArrayOf(3, 0, 6, 1, 5)
        assertThat(solution.hIndex(citations)).isEqualTo(3)
    }

    @Test
    fun test2() {
        val citations = intArrayOf(1, 3, 1)
        assertThat(solution.hIndex(citations)).isEqualTo(1)
    }

    @Test
    fun test3() {
        val citations = intArrayOf(0, 0, 1)
        assertThat(solution.hIndex(citations)).isEqualTo(1)
    }

    @Test
    fun test4() {
        val citations = intArrayOf(0, 0, 0)
        assertThat(solution.hIndex(citations)).isEqualTo(0)
    }

}